db/firebase/parser.js

```diff
@@ -0,0 +1,101 @@
+// Firebase requirements
+require("firebase/firestore");
+const csv = require('csv-parser')
+const fs = require('fs')
+require('dotenv').config()
+const admin = require('firebase-admin');
+
+// initialize firebase
+function initializeFirebaseAdmin(adminSDK, databaseURL) {
+    return admin.initializeApp({
+        credential: admin.credential.cert(adminSDK),
+        databaseURL: databaseURL,
+    });
+}
+const adminSDK = JSON.parse(process.env.NWPLUSADMINSDK);
+let app = initializeFirebaseAdmin(adminSDK, "https://nwplus-bot.firebaseio.com");
+
+// save second argument as type of members to add
+let type = process.argv[2];
+if (type == undefined) {
+    throw new Error('no defined type!')
+}
+// optional third argument; if "true", all their previous types will be overwritten by this new one
+let overwrite = false;
+if (process.argv[3] === 'true') {
+    overwrite = true;
+};
+
+class Registration {
+    constructor(email) {
+        this.email = email
+        this.types = [];
+    }
+}
+
+const results = []
+const all_regs = {}
+fs.createReadStream('registrations.csv') // requires a registrations.csv file in root directory to run
+    .pipe(csv())
+    .on('data', (data) => results.push(data))
+    .on('end', () => {
+        const processed = results.map((row) => {
+            const email = row['What is your primary email that can we contact you with?'] || row['Email Address']
+
+            const r = new Registration(email)
+            return r
+        })
+
+        processed.forEach(r => {
```

Comment:
Could we use a map here? I think maps are faster and would be easier to use? Easier to iterate over?

Reply:
This was from the original implementation so I didn't change it. I think maps aren't inherently faster than arrays but I could just get rid of the processed array altogether